mixer/adapter/circonus/circonus.go

```diff
@@ -0,0 +1,196 @@
+				SubmissionURL: b.adpCfg.SubmissionUrl,
+			},
+		},
+		Log:      log.New(bridge, "", 0),
+		Debug:    true, // enable [DEBUG] level logging for env.Logger
+		Interval: "0s", // flush via ScheduleDaemon based ticker
+	}
+
+	cm, err := cgm.NewCirconusMetrics(cmc)
+	if err != nil {
+		err = env.Logger().Errorf("Could not create NewCirconusMetrics: %v", err)
+		return nil, err
+	}
+
+	env.ScheduleDaemon(
+		func() {
+			for range time.NewTicker(b.adpCfg.SubmissionInterval).C {
+				cm.Flush()
+			}
+		})
+
+	metrics := make(map[string]config.Params_MetricInfo_Type)
+	ac := b.adpCfg
+	for _, adpMetric := range ac.Metrics {
+		metrics[adpMetric.Name] = adpMetric.Type
+	}
+
+	return &handler{cm: cm, env: env, metrics: metrics}, nil
+}
+
+// SetAdapterConfig assigns operator configuration to the builder
+func (b *builder) SetAdapterConfig(cfg adapter.Config) {
+	b.adpCfg = cfg.(*config.Params)
+}
+
+// Validate checks the SubmissionUrl parameter is well formed, and needed metrics are declared
+func (b *builder) Validate() (ce *adapter.ConfigErrors) {
+
+	// verify the submission url is well formed
+	if _, err := url.ParseRequestURI(b.adpCfg.SubmissionUrl); err != nil {
+		ce = ce.Append("submission_url", err)
+	}
+
+	// put the metric config into a map for use below
+	configMetrics := make(map[string]config.Params_MetricInfo_Type)
+	for _, configMetric := range b.adpCfg.Metrics {
+		configMetrics[configMetric.Name] = configMetric.Type
+	}
+
+	// verify there are no metric types without a corresponding config item
+	for metricName := range b.metricTypes {
+		if _, ok := configMetrics[metricName]; !ok {
+			err := fmt.Errorf("missing metric configuration %v", metricName)
+			ce = ce.Append("metrics", err)
+		}
+	}
+
+	// verify there are no config items without a corresponding metric type
+	for configMetric := range configMetrics {
+		if _, ok := b.metricTypes[configMetric]; !ok {
+			err := fmt.Errorf("missing metric type for %v", configMetric)
+			ce = ce.Append("metrics", err)
+		}
+	}
+
+	// because SubmissionInterval is a duration type, validation errors will be caught before this point
+
+	return
+}
+
+// SetMetricTypes sets the available metric types in the builder
+func (b *builder) SetMetricTypes(types map[string]*metric.Type) {
+	b.metricTypes = types
+}
+
+// HandleMetric submits metrics to Circonus via circonus-gometrics
+func (h *handler) HandleMetric(ctx context.Context, insts []*metric.Instance) error {
+
+	for _, inst := range insts {
+
+		metricName := inst.Name
+		metricType := h.metrics[metricName]
+
+		switch metricType {
+
+		case config.GAUGE:
+			value, _ := inst.Value.(int64)
+			h.cm.Gauge(metricName, value)
+
+		case config.COUNTER:
+			h.cm.Increment(metricName)
+
+		case config.DISTRIBUTION:
+			value, _ := inst.Value.(time.Duration)
+			h.cm.Timing(metricName, float64(value))
+		}
+
+	}
+
+	return nil
+}
+
+// Close flushes any buffered metrics
+func (h *handler) Close() error {
+	h.cm.Flush()
```

Comment:
Uh ho, forgot to clean up the daemon your started up there. That bad boy needs to shut down...

Reply:
Thanks - implementing a solution for that. Should I expect that `Close()` can be called at most once, or possibly multiple times?